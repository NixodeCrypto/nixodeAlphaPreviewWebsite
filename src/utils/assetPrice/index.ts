import assetPrice from './assetPrice';

export default assetPrice;
