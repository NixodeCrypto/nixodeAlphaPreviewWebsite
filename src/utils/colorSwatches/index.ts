import colorSwatches from './colorSwatches';

export * from './colorSwatches';
export default colorSwatches;
