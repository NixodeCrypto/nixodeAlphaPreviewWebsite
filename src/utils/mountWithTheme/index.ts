import mountWithTheme from './mountWithTheme';

export default mountWithTheme;
