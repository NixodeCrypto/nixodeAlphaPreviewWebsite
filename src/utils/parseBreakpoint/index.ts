import parseBreakpoint from './parseBreakpoint';

export default parseBreakpoint;
