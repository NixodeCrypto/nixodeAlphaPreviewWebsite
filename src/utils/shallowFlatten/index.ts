import shallowFlatten from './shallowFlatten';

export default shallowFlatten;
