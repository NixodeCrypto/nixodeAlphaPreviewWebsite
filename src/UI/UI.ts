import { css, Theme } from '@emotion/react';
import { rem } from 'polished';
import colorSwatches from '@/utils/colorSwatches'; // import directly to avoid circular dependency with utils/token

export const GlobalTheme: Theme = {
  space: {
    '-xl': rem(-64),
    '-lg': rem(-32),
    '-md': rem(-24),
    '-sm': rem(-16),
    '-xs': rem(-8),
    '-xss': rem(-4),
    xss: rem(4),
    xs: rem(8),
    sm: rem(16),
    md: rem(24),
    lg: rem(32),
    xl: rem(64),
  },
  fontSizes: {
    // Footer fs
    footer: rem(10),
    // Caption fs
    captionSm: rem(11),
    captionLg: rem(12),
    // Body fs
    bodySm: rem(14),
    bodyLg: rem(16),
    // H6 fs
    h6Sm: rem(12),
    h6: rem(14),
    // H5 fs
    h5Sm: rem(14),
    h5: rem(16),
    // H4 fs
    h4Sm: rem(16),
    h4: rem(20),
    // H3 fs
    h3Sm: rem(20),
    h3: rem(24),
    // H2 fs
    h2Sm: rem(24),
    h2: rem(34),
    // H1 fs
    h1Sm: rem(34),
    h1: rem(52),
  },
  colors: {
    // Brand Colors
    primary: colorSwatches('#FD9500'),
    secondary: colorSwatches('#82B5B2'),
    accent: colorSwatches('#DC5D3C'),
    // Base Colors
    red: colorSwatches('#f44336'),
    pink: colorSwatches('#e91e63'),
    purple: colorSwatches('#9c27b0'),
    deepPurple: colorSwatches('#673ab7'),
    indigo: colorSwatches('#3f51b5'),
    blue: colorSwatches('#2196f3'),
    lightBlue: colorSwatches('#03a9f4'),
    cyan: colorSwatches('#00bcd4'),
    teal: colorSwatches('#009688'),
    green: colorSwatches('#4caf50'),
    lightGreen: colorSwatches('#8bc34a'),
    lime: colorSwatches('#cddc39'),
    yellow: colorSwatches('#ffeb3b'),
    amber: colorSwatches('#ffc107'),
    orange: colorSwatches('#ff9800'),
    deepOrange: colorSwatches('#ff5722'),
    brown: colorSwatches('#795548'),
    grey: colorSwatches('#9e9e9e'),
    blueGrey: colorSwatches('#607d8b'),
  },
  fonts: {
    title: 'Inter, Roboto, sans-serif',
    text: 'Inter, Roboto, sans-serif',
  },
  fontWeights: {
    thin: 100,
    light: 300,
    regular: 400,
    medium: 500,
    semiBold: 600,
    bold: 700,
    extraBold: 800,
  },
  lineHeights: {},
  letterSpacings: {
    sm: rem(1.5),
  },
  sizes: {
    xss: rem(14),
    xs: rem(19),
    sm: rem(28),
    md: rem(37),
    lg: rem(46),
    xl: rem(64),
    xxl: rem(144),
    card: rem(215),
    cardXl: rem(225),
    maxLayoutSm: rem(320),
    maxLayoutMd: rem(400),
    img: rem(432),
    imgXl: rem(480),
    input: rem(280),
    inputXl: rem(480),
    max: '100%',
  },
  screenMaxWidths: {
    xs: rem(400),
    sm: rem(700),
    md: rem(1100),
  },
  breakpoints: {
    xss: '0px',
    xs: '320px',
    sm: '640px',
    md: '960px',
    lg: '1280px',
  },
  borders: {
    sm: `${rem(2)} solid`,
  },
  radii: {
    sm: rem(4),
    xl: rem(24),
  },
  shadows: {
    focus: '0 0 0 3px rgba(66,153,225,0.3)',
    reg: 'rgba(0,0,0,0.1) 0px 4px 12px',
  },
  zIndices: {
    navbar: 9999,
    popup: 6000,
  },
  transition: {
    standard: 'all 0.1s ease-in-out, width 0s, height 0s',
  },
};

export const GlobalReset = css`
  html {
    line-height: 1.15; /* 1 */
    -webkit-text-size-adjust: 100%; /* 2 */
    scroll-behavior: smooth;
  }
  body {
    margin: 0;
  }
  main {
    display: block;
  }
  h1 {
    font-size: 2em;
    margin: 0 0;
  }
  hr {
    box-sizing: content-box; /* 1 */
    height: 0; /* 1 */
    overflow: visible; /* 2 */
  }
  pre {
    font-family: monospace, monospace; /* 1 */
    font-size: 1em; /* 2 */
  }
  a {
    background-color: transparent;
  }
  abbr[title] {
    border-bottom: none; /* 1 */
    text-decoration: underline; /* 2 */
    text-decoration: underline dotted; /* 2 */
  }
  b,
  strong {
    font-weight: bolder;
  }
  code,
  kbd,
  samp {
    font-family: monospace, monospace; /* 1 */
    font-size: 1em; /* 2 */
  }
  small {
    font-size: 80%;
  }
  sub,
  sup {
    font-size: 75%;
    line-height: 0;
    position: relative;
    vertical-align: baseline;
  }
  sub {
    bottom: -0.25em;
  }
  sup {
    top: -0.5em;
  }
  img {
    border-style: none;
  }
  button,
  input,
  optgroup,
  select,
  textarea {
    font-family: inherit; /* 1 */
    font-size: 100%; /* 1 */
    line-height: 1.15; /* 1 */
    margin: 0; /* 2 */
  }
  button,
  input {
    /* 1 */
    overflow: visible;
  }
  button,
  select {
    /* 1 */
    text-transform: none;
  }
  button,
  [type='button'],
  [type='reset'],
  [type='submit'] {
    -webkit-appearance: button;
  }
  button::-moz-focus-inner,
  [type='button']::-moz-focus-inner,
  [type='reset']::-moz-focus-inner,
  [type='submit']::-moz-focus-inner {
    border-style: none;
    padding: 0;
  }
  button:-moz-focusring,
  [type='button']:-moz-focusring,
  [type='reset']:-moz-focusring,
  [type='submit']:-moz-focusring {
    outline: 1px dotted ButtonText;
  }
  fieldset {
    padding: 0.35em 0.75em 0.625em;
  }
  legend {
    box-sizing: border-box; /* 1 */
    color: inherit; /* 2 */
    display: table; /* 1 */
    max-width: 100%; /* 1 */
    padding: 0; /* 3 */
    white-space: normal; /* 1 */
  }
  progress {
    vertical-align: baseline;
  }
  textarea {
    overflow: auto;
  }
  [type='checkbox'],
  [type='radio'] {
    box-sizing: border-box; /* 1 */
    padding: 0; /* 2 */
  }
  [type='number']::-webkit-inner-spin-button,
  [type='number']::-webkit-outer-spin-button {
    height: auto;
  }
  [type='search'] {
    -webkit-appearance: textfield; /* 1 */
    outline-offset: -2px; /* 2 */
  }
  [type='search']::-webkit-search-decoration {
    -webkit-appearance: none;
  }
  ::-webkit-file-upload-button {
    -webkit-appearance: button; /* 1 */
    font: inherit; /* 2 */
  }
  details {
    display: block;
  }
  summary {
    display: list-item;
  }
  template {
    display: none;
  }
  [hidden] {
    display: none;
  }
`;
