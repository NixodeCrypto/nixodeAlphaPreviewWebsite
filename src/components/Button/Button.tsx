/* @jsxImportSource @emotion/react */
import React from 'react';
import styled from '@emotion/styled';
import { Theme, css } from '@emotion/react';
import { transparentize } from 'polished';
import {
  space,
  SpaceProps,
  layout,
  LayoutProps,
  border,
  BorderProps,
  variant as CSSVariant,
} from '@/system';
import { ColorSwatches } from '@/utils/colorSwatches';
import { token } from '@/utils';

export interface IProps
  extends ButtonProps,
    SpaceProps,
    LayoutProps,
    BorderProps {
  color?: 'primary' | 'secondary' | 'accent' | 'grey';
  size?: 'sm' | 'md' | 'lg' | 'icon';
  variant?: 'solid' | 'text' | 'outlined';
}

const solidGreyComposer = (color: keyof Theme['colors']) => {
  const tokenColorPicks =
    color === 'grey'
      ? ['100', '200', '300', '800']
      : ['500', '600', '700', 'white'];

  for (let i = 0; i < tokenColorPicks.length; i += 1) {
    tokenColorPicks[i] =
      token.colors(
        `${color}.${tokenColorPicks[i]}` as `${keyof Theme['colors']}.${keyof ColorSwatches}`,
      ) ?? tokenColorPicks[i];
  }

  return css`
    border-color: ${tokenColorPicks[0]};
    background: ${tokenColorPicks[0]};
    color: ${tokenColorPicks[3]};
    &:hover, &:focus-visible {
      border-color: ${tokenColorPicks[1]};
      background: ${tokenColorPicks[1]};
    }
    &:active {
      border-color: ${tokenColorPicks[2]};
      background: ${tokenColorPicks[2]};
  `;
};

const ButtonRoot = styled.button<IProps>`
  position: relative;
  font-family: ${token.fonts('text')};
  font-weight: ${token.fontWeights('semiBold')};
  cursor: pointer;
  border: ${token.borders('sm')};
  border-color: transparent;
  border-radius: ${token.radii('sm')};
  transition: ${token.transition('standard')};
  white-space: nowrap;
  box-sizing: border-box;
  background: transparent;
  display: flex;
  justify-content: center;
  align-items: center;

  &:focus-visible {
    outline: 0;
  }

  ${(props) =>
    props.color &&
    CSSVariant(props.variant, {
      all: css`
        color: ${token.colors(`${props.color}.500`)};
      `,
      solid: solidGreyComposer(props.color),
      text: css`
        &:hover,
        &:focus-visible {
          background: ${transparentize(
            0.9,
            token.colors(`${props.color}.500`),
          )};
        }
        &:active {
          background: ${transparentize(
            0.7,
            token.colors(`${props.color}.500`),
          )};
          color: ${token.colors(`${props.color}.600`)};
        }
      `,
      outlined: css`
        border: ${token.borders('sm')};
        border-color: ${token.colors(`${props.color}.500`)};
        &:hover,
        &:focus-visible {
          background: ${transparentize(
            0.9,
            token.colors(`${props.color}.500`),
          )};
        }
        &:active {
          background: ${transparentize(
            0.7,
            token.colors(`${props.color}.500`),
          )};
          border-color: ${token.colors(`${props.color}.600`)};
          color: ${token.colors(`${props.color}.600`)};
        }
      `,
    })}

  ${(props) =>
    CSSVariant(props.size, {
      all: css`
        font-size: ${token.fontSizes('bodyLg')};
      `,
      icon: css`
        height: ${token.sizes('sm')};
      `,
      sm: css`
        height: ${token.sizes('sm')};
        padding-left: ${token.space('xs')};
        padding-right: ${token.space('xs')};
        font-size: ${token.fontSizes('bodySm')};
      `,
      md: css`
        height: ${token.sizes('md')};
        padding-left: ${token.space('sm')};
        padding-right: ${token.space('sm')};
      `,
      lg: css`
        height: ${token.sizes('lg')};
        padding-left: ${token.space('md')};
        padding-right: ${token.space('md')};
      `,
    })}

  ${space};
  ${layout};
  ${border};
`;

const Button = React.forwardRef(
  (props: IProps, ref: React.Ref<HTMLButtonElement>): JSX.Element => {
    const {
      children,
      type = 'button',
      color = 'primary',
      variant = 'solid',
      size = 'md',
      ...other
    } = props;

    return (
      <ButtonRoot
        type={type}
        color={color}
        size={size}
        ref={ref}
        variant={variant}
        {...other}
      >
        {children}
      </ButtonRoot>
    );
  },
);

Button.displayName = 'Button';
export default Button;
