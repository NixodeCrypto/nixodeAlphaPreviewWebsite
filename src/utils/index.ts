import variant from '@/utils/variant';
import colorSwatches from '@/utils/colorSwatches';
import token from '@/utils/token';
import mq from '@/utils/mq';
import parseBreakpoint from '@/utils/parseBreakpoint';
import mountWithTheme from '@/utils/mountWithTheme';
import strToObj from '@/utils/strToObj';

export {
  token,
  variant,
  colorSwatches,
  mq,
  parseBreakpoint,
  mountWithTheme,
  strToObj,
};
