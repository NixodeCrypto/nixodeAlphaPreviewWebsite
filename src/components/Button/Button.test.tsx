import { cleanup } from '@testing-library/react';
import { matchers } from '@emotion/jest';
import { transparentize } from 'polished';
import Button from '.';
import { GlobalTheme } from '@/UI';
import { mountWithTheme } from '@/utils';

expect.extend(matchers);

beforeEach(cleanup);

describe('components/Button', () => {
  describe('standard', () => {
    it('base functions', () => {
      const mockFn = jest.fn();
      const wrapper = mountWithTheme(<Button onClick={mockFn} />);
      wrapper.simulate('click');
      expect(mockFn).toHaveBeenCalled();
    });
    it('snapshot testing', () => {
      const wrapper = mountWithTheme(<Button />);
      expect(wrapper).toMatchSnapshot();
    });
  });
  describe('color', () => {
    describe('primary button', () => {
      it('correct background and color', () => {
        const wrapper = mountWithTheme(<Button color="primary" />);

        // correct background
        expect(wrapper).toHaveStyleRule(
          'background',
          GlobalTheme.colors.primary[500],
        );

        // correct color
        expect(wrapper).toHaveStyleRule('color', 'white');
      });
      it('primary &:hover state', () => {
        const wrapper = mountWithTheme(<Button color="primary" />);

        expect(wrapper).toHaveStyleRule(
          'background',
          GlobalTheme.colors.primary[600],
          {
            target: ':hover',
          },
        );
      });
      it('primary &:active state', () => {
        const wrapper = mountWithTheme(<Button color="primary" />);
        expect(wrapper).toHaveStyleRule(
          'background',
          GlobalTheme.colors.primary[700],
          {
            target: ':active',
          },
        );
      });
    });

    describe('secondary button', () => {
      it('correct background and color', () => {
        const wrapper = mountWithTheme(<Button color="secondary" />);

        // correct background
        expect(wrapper).toHaveStyleRule(
          'background',
          GlobalTheme.colors.secondary[500],
        );

        // correct color
        expect(wrapper).toHaveStyleRule('color', 'white');
      });
      it('secondary &:hover state', () => {
        const wrapper = mountWithTheme(<Button color="secondary" />);

        expect(wrapper).toHaveStyleRule(
          'background',
          GlobalTheme.colors.secondary[600],
          {
            target: ':hover',
          },
        );
      });
      it('secondary &:active state', () => {
        const wrapper = mountWithTheme(<Button color="secondary" />);
        expect(wrapper).toHaveStyleRule(
          'background',
          GlobalTheme.colors.secondary[700],
          {
            target: ':active',
          },
        );
      });
    });

    describe('accent button', () => {
      it('correct background and color', () => {
        const wrapper = mountWithTheme(<Button color="accent" />);

        // correct background
        expect(wrapper).toHaveStyleRule(
          'background',
          GlobalTheme.colors.accent[500],
        );

        // correct color
        expect(wrapper).toHaveStyleRule('color', 'white');
      });
      it('accent &:hover state', () => {
        const wrapper = mountWithTheme(<Button color="accent" />);

        expect(wrapper).toHaveStyleRule(
          'background',
          GlobalTheme.colors.accent[600],
          {
            target: ':hover',
          },
        );
      });
      it('accent &:active state', () => {
        const wrapper = mountWithTheme(<Button color="accent" />);
        expect(wrapper).toHaveStyleRule(
          'background',
          GlobalTheme.colors.accent[700],
          {
            target: ':active',
          },
        );
      });
    });

    describe('grey button', () => {
      it('correct background and color', () => {
        const wrapper = mountWithTheme(<Button color="grey" />);

        // correct background
        expect(wrapper).toHaveStyleRule(
          'background',
          GlobalTheme.colors.grey[100],
        );

        // correct color
        expect(wrapper).toHaveStyleRule('color', GlobalTheme.colors.grey[800]);
      });
      it('grey &:hover state', () => {
        const wrapper = mountWithTheme(<Button color="grey" />);

        expect(wrapper).toHaveStyleRule(
          'background',
          GlobalTheme.colors.grey[200],
          {
            target: ':hover',
          },
        );
      });
      it('grey &:active state', () => {
        const wrapper = mountWithTheme(<Button color="grey" />);
        expect(wrapper).toHaveStyleRule(
          'background',
          GlobalTheme.colors.grey[300],
          {
            target: ':active',
          },
        );
      });
    });
  });

  describe('variants', () => {
    it('solid variant', () => {
      const wrapper = mountWithTheme(<Button variant="solid" />);
      expect(wrapper).toHaveStyleRule('border', '0');

      expect(wrapper).toHaveStyleRule(
        'background',
        GlobalTheme.colors.primary[500],
      );
    });
    it('text variant', () => {
      const wrapper = mountWithTheme(<Button variant="text" />);
      expect(wrapper).toHaveStyleRule('background', 'transparent');
      expect(wrapper).toHaveStyleRule(
        'background',
        transparentize(0.9, GlobalTheme.colors.primary[500]),
        {
          target: ':hover',
        },
      );

      expect(wrapper).toHaveStyleRule(
        'background',
        transparentize(0.7, GlobalTheme.colors.primary[500]),
        {
          target: ':active',
        },
      );

      expect(wrapper).toHaveStyleRule(
        'color',
        GlobalTheme.colors.primary[600],
        {
          target: ':active',
        },
      );
    });

    it('outlined variant', () => {
      const wrapper = mountWithTheme(<Button variant="outlined" />);

      expect(wrapper).toHaveStyleRule('background', 'transparent');
      expect(wrapper).toHaveStyleRule('border', GlobalTheme.borders.sm);
      expect(wrapper).toHaveStyleRule('color', GlobalTheme.colors.primary[500]);

      expect(wrapper).toHaveStyleRule(
        'background',
        transparentize(0.9, GlobalTheme.colors.primary[500]),
        {
          target: ':hover',
        },
      );

      expect(wrapper).toHaveStyleRule(
        'background',
        transparentize(0.7, GlobalTheme.colors.primary[500]),
        {
          target: ':active',
        },
      );

      expect(wrapper).toHaveStyleRule(
        'color',
        GlobalTheme.colors.primary[600],
        {
          target: ':active',
        },
      );

      expect(wrapper).toHaveStyleRule(
        'border-color',
        GlobalTheme.colors.primary[600],
        {
          target: ':active',
        },
      );
    });
  });

  describe('sizes', () => {
    it('icon', () => {
      const wrapper = mountWithTheme(<Button size="icon" />);
      expect(wrapper).toHaveStyleRule('height', GlobalTheme.sizes.sm);
    });

    it('sm', () => {
      const wrapper = mountWithTheme(<Button size="sm" />);
      expect(wrapper).toHaveStyleRule('height', GlobalTheme.sizes.sm);
      expect(wrapper).toHaveStyleRule('padding', GlobalTheme.space.xs);
      expect(wrapper).toHaveStyleRule(
        'font-size',
        GlobalTheme.fontSizes.bodySm,
      );
    });

    it('md', () => {
      const wrapper = mountWithTheme(<Button size="md" />);
      expect(wrapper).toHaveStyleRule('height', GlobalTheme.sizes.md);
      expect(wrapper).toHaveStyleRule('padding', GlobalTheme.space.sm);
      expect(wrapper).toHaveStyleRule(
        'font-size',
        GlobalTheme.fontSizes.bodyLg,
      );
    });

    it('lg', () => {
      const wrapper = mountWithTheme(<Button size="lg" />);
      expect(wrapper).toHaveStyleRule('height', GlobalTheme.sizes.lg);
      expect(wrapper).toHaveStyleRule('padding', GlobalTheme.space.md);
      expect(wrapper).toHaveStyleRule(
        'font-size',
        GlobalTheme.fontSizes.bodyLg,
      );
    });
  });

  describe('maxWidth', () => {
    it('makes width 100%', () => {
      const wrapper = mountWithTheme(<Button maxWidth />);
      expect(wrapper).toHaveStyleRule('width', '100%');
    });
  });
});
