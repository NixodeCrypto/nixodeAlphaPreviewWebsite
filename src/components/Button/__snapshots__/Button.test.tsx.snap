// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`components/Button standard matches snapshot 1`] = `ReactWrapper {}`;
